.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Dynatrace LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build and test
+on:
+  pull_request:
+    branches: [ "main" ]
+  push:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Installing dependencies
+        run: pip install ansible
+      - name: Building the collection
+        run: ansible-galaxy collection build .
+      - name: Installing the collection
+        run: ansible-galaxy collection install dynatrace-oneagent*
+      - name: Running sanity test
+        run: pushd ~/.ansible/collections/ansible_collections/dynatrace/oneagent && ansible-test sanity && popd
+      - uses: actions/upload-artifact@v4
```

Comment:
upload should be done at least as a separate step

Reply:
What is the reason for that? I don't think putting it to separate job and create dependency between uploading and building just for a sake of separating them has any meaning